spring-cloud-azure-keyvault-config/src/main/java/com/microsoft/azure/spring/cloud/keyvault/config/auth/AadKeyVaultCredentials.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.spring.cloud.keyvault.config.auth;
+
+import com.microsoft.aad.adal4j.AuthenticationContext;
+import com.microsoft.aad.adal4j.AuthenticationResult;
+import com.microsoft.azure.keyvault.authentication.KeyVaultCredentials;
+import lombok.AllArgsConstructor;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Subclass of {@link KeyVaultCredentials}.
+ */
+@AllArgsConstructor
+public class AadKeyVaultCredentials extends KeyVaultCredentials {
+    // TODO: make it configurable.
+    private static final int TIMEOUT_IN_SECONDS = 15;
+
+    private AuthenticationExecutor authExecutor;
+
+    @Override
+    public String doAuthenticate(String authorization, String resource, String scope) {
+        AuthenticationResult result;
+
+        //Starts a service to fetch access token.
+        ExecutorService service = null;
+        try {
+            service = Executors.newSingleThreadExecutor();
+            AuthenticationContext context = new AuthenticationContext(authorization, false, service);
+            result = authExecutor.acquireToken(context, resource).get(TIMEOUT_IN_SECONDS, TimeUnit.SECONDS);
+        } catch (Exception ex) {

```

Comment:
1. How about catching specific exception here?
2. Maybe we can have a dedicated thread pool for such auth. For each method call, creating and destroying a thread pool looks too heavy.

Reply:
> 1. How about catching specific exception here?

It is doable. Just curious any concern that we want to make it specific? We actually don't care what exception it is and will re-throw it with a clearer error message.

> 2. Maybe we can have a dedicated thread pool for such auth. For each method call, creating and destroying a thread pool looks too heavy.

Sounds good. Let's make the thread pool a private field of `AadKeyVaultCredentials`.

Reply:
Normally, we only handle checked exception when we know how to deal with. For other runtime exception, just let them throw.

Reply:
I see. Then in this case I think it is okay to catch all exceptions because we want to provide better error message to users. A run-time exception with deep stack trace and very specific error messages will cost user more time to investigate.
Thoughts?